### PR TITLE
Animation:  interactive zoom/pan with blitting does not work

### PR DESCRIPTION
**Pull request by onatt**
## PR Summary
In an animation using blitting that contains some static graphics elements, the static parts of the figure are not updated after an interactive zoom/pan event. However, they are correctly updated after an resize event (see issue #13478).

## PR Checklist

- [ ] Has Pytest style unit tests
- [X] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
This seems OK.  Is the test just to run what you did in #13478?  It would be good if a blitting expert chimed in.  Assigning one ;-)

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -1222,6 +1222,14 @@ def _setup_blit(self):
         # axes
         self._blit_cache = dict()
         self._drawn_artists = []
+        def remove_from_cache(ax):
+            try:
+                del self._blit_cache[ax]
```
Comment (dopplershift) [inline lib/matplotlib/animation.py]: Could replace the whole block with:
```python
self._blit_cache.pop(ax, None)
```

Comment (onatt) [reply lib/matplotlib/animation.py]: @dopplershift,
I really like your proposal. The second argument of dict.pop did not come to my mind. We could make my proposal much more concise:
```
        self._drawn_artists = []
        for ax in self._fig.axes:
            ax.callbacks.connect('xlim_changed',
                                 lambda ax: self._blit_cache.pop(ax, None))
            ax.callbacks.connect('ylim_changed',
                                 lambda ax: self._blit_cache.pop(ax, None))
        self._resize_id = self._fig.canvas.mpl_connect('resize_event',
                                                       self._handle_resize)
```
This was my first contribution to matplotlib. Shall I file another pull request for this modification?

Comment (jklymak) [reply lib/matplotlib/animation.py]: @onatt, no just modify this PR, add a new commit on your machine, and then push...  https://matplotlib.org/devel/index.html

**Comment by timhoffm** [comment]
As this is a bugfix, should it go in 3.1?

**Comment by dopplershift** [comment]
I think that would be appropriate. I think 3.1 branched after it was submitted.

**Pull request merged**